src/com/esotericsoftware/kryo/GenericsResolver.java

```diff
@@ -0,0 +1,34 @@
+package com.esotericsoftware.kryo;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/** Helper class that resolves a type name variable to a concrete class using the current class serialization stack
+ *
+ * @author Jeroen van Erp <jeroen@hierynomus.com> */
+public class GenericsResolver {
```

Comment:
Yes, I would agree, _but_ `FieldSerializer` and `FieldSerializerGenericsUtil` live in a different package than `Kryo`, so the class itself needs to be public. I'll restrict the methods in there.